XRMSolutionDependencyChecker/PluginControl.cs

```diff
@@ -150,24 +181,161 @@ public override void UpdateConnection(IOrganizationService newService, Connectio
         /// <param name="e"></param>
         private void LoadSolution_Button_Click(object sender, EventArgs e)
         {
+            byte[] SolutionFile = null;
+
             OpenSolution.ShowDialog();
 
             if (OpenSolution.SafeFileName == "OpenSolution")
                 return;
 
-            byte[] SolutionFile = File.ReadAllBytes(OpenSolution.FileName);
+            // If file isn't a zip, stop here
+            if (!OpenSolution.SafeFileName.Contains(".zip"))
+            {
+                outputText.Text = "Error: Expected solution is not a .zip";
+                outputText.Visible = true;
+                return;
+            }
 
-            BackgroundProgressIndicator.RunWorkerAsync(SolutionFile);
+            WorkAsync(new WorkAsyncInfo
+            {
+                Message = "Checking for dependencies...",
+                IsCancelable = true,
+                Work = (worker, eventargs) =>
+                {
+                    SolutionFile = File.ReadAllBytes(OpenSolution.FileName);
```

Comment:
Why are you reading all the bytes here again when we read them above?

Reply:
We are not reading them above anymore. The read is being done inside the WorkAsync as of now